airbyte-integrations/connectors/source-amplitude/source_amplitude/components.py

```diff
@@ -0,0 +1,111 @@
+    def extract_records(self, response: requests.Response) -> List[Record]:
+        response_data = response.json().get("data", [])
+        if response_data:
+            # From the Amplitude documentation it follows that "series" is an array with one element which is itself
+            # an array that contains the average session length for each day.
+            # https://developers.amplitude.com/docs/dashboard-rest-api#returns-2
+            series = response_data.get("series", [])
+            if len(series) > 0:
+                series = series[0]  # get the nested list
+                return [{"date": date, "length": series[i]} for i, date in enumerate(response_data["xValues"])]
+        return []
+
+
+class ActiveUsersRecordExtractor(RecordExtractor):
+    def extract_records(self, response: requests.Response) -> List[Record]:
+        response_data = response.json().get("data", [])
+        if response_data:
+            series = list(map(list, zip(*response_data["series"])))
+            if series:
+                return [
+                    {"date": date, "statistics": dict(zip(response_data["seriesLabels"], series[i]))}
+                    for i, date in enumerate(response_data["xValues"])
+                ]
+        return []
+
+
+@dataclass
+class EventsExtractor(RecordExtractor, JsonSchemaMixin):
+    config: Config
+    options: InitVar[Mapping[str, Any]]
+
+    def __post_init__(self, options: Mapping[str, Any]):
+        self.name = options.get("name")
+
+    def _get_schema_root_properties(self):
+        schema_loader = JsonFileSchemaLoader(config=self.config, options={"name": self.name})
+        schema = schema_loader.get_json_schema()
+        return schema["properties"]
+
+    def _get_date_time_items_from_schema(self):
+        """
+        Get all properties from schema with format: 'date-time'
+        """
+        result = []
+        schema = self._get_schema_root_properties()
+        for key, value in schema.items():
+            if value.get("format") == "date-time":
+                result.append(key)
+        return result
+
+    def _date_time_to_rfc3339(self, record: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        """
+        Transform 'date-time' items to RFC3339 format
+        """
+        date_time_fields = self._get_date_time_items_from_schema()
+        for item in record:
+            if item in date_time_fields:
+                dt_value = record[item]
+                if not dt_value:
+                    # either null or empty string, leave it as it
+                    record[item] = dt_value
+                else:
+                    record[item] = pendulum.parse(dt_value).to_rfc3339_string()
+        return record
+
+    def extract_records(
+        self,
+        response: requests.Response,
+    ) -> List[Record]:
+        try:
+            zip_file = zipfile.ZipFile(io.BytesIO(response.content))
+        except zipfile.BadZipFile as e:
+            logger.exception(e)
```

Comment:
Why both `logger.exception` and `logger.error`? Couldn't we just do this? 
```
            logger.exception(
                f"Received an invalid zip file in response to URL: {response.request.url}."
                f"The size of the response body is: {len(response.content)}"
            )
```

Reply:
updated